graphcast/data_utils.py

```diff
@@ -270,23 +240,43 @@ def extract_input_target_times(
   (target_lead_times, target_duration
    ) = _process_target_lead_times_and_get_duration(target_lead_times)
 
-  # Shift the coordinates for the time axis so that a timedelta of zero
-  # corresponds to the forecast reference time. That is, the final timestep
-  # that's available as input to the forecast, with all following timesteps
-  # forming the target period which needs to be predicted.
-  # This means the time coordinates are now forecast lead times.
+  input_duration = pd.Timedelta(input_duration)
   time = dataset.coords["time"]
-  dataset = dataset.assign_coords(time=time + target_duration - time[-1])
 
-  # Slice out targets:
-  targets = dataset.sel({"time": target_lead_times})
+  # Slice out inputs and targets:
+  if justify == 'left':
+    # Inputs correspond to the first time elements within the input duration
+    # Targets follow immediatly after per the target lead times
+    target_start_time = int(input_duration.total_seconds()/3600/6)
+    target_end_time = int(input_duration.total_seconds()/3600/6) + int(target_duration.total_seconds()/3600/6)
+
+    inputs = dataset.isel(time=slice(int(target_start_time)))
+    inputs['time'] = inputs['time'] - input_duration + time[1]  
+
+    targets = dataset.isel(time=slice(target_start_time,target_end_time))
```

Comment:
Please take a look at the google python style guide. Some general comments:
- There should be spaces between function arguments, no trailing spaces, etc.
- Comments are in english and should use punctuation (eg missing periods above).
- Remove debugging statements (commented out code, print statements below).